CMakeLists.txt

```diff
@@ -114,13 +137,15 @@ zephyr_compile_options(
   $<$<COMPILE_LANGUAGE:ASM>:-D_ASMLANGUAGE>
 )
 
+if(NOT CONFIG_ARCH_POSIX)
 zephyr_ld_options(
   -nostartfiles
   -nodefaultlibs
   -nostdlib
   -static
   -no-pie
-  )
+)
+endif()
```

Comment:
You need to get rid of this if statement for the same reasons as above, I assume this can easily be done with at least one of the techniques mentioned above.